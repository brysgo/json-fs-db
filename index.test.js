const path = require("path");
const fs = require("fs");

const { get, set, link } = require("./")(path.resolve("./fixtures"));

test("getting simple attribute from json", async () => {
  expect(await get("foo.stuff")).toBe("things");
});

test("getting an attribute that lives in a file in a folder", async () => {
  expect(await get("blah.baz.nested.simple")).toBe("object");
});

test("setting an attribute in a json file", async () => {
  expect(await get("blah.baz.nested.justSet")).toBe(undefined);
  await set("blah.baz.nested.justSet", "to something");
  expect(await get("blah.baz.nested.justSet")).toBe("to something");
  await set("blah.baz.nested.justSet", undefined);
  expect(await get("blah.baz.nested.justSet")).toBe(undefined);
  expect(fs.readFileSync("./fixtures/blah/baz.js").toString('utf8')).toMatchSnapshot();
});

test("relational links between models", async () => {
  expect(await get("foo.bar.testLink")).toBe(undefined);
  await set("foo.bar.testLink", link("blah.baz"));
  expect(await get("foo.bar.testLink.confirmed")).toBe("here it is");
  await set("foo.bar.testLink", undefined);
  expect(await get("foo.bar.testLink")).toBe(undefined);
});
