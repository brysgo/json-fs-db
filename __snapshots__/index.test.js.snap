// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`setting an attribute in a json file 1`] = `
"const { link } = require(\\"json-fs-db\\")(\\"../../\\");
module.exports = {
  \\"nested\\": {
    \\"simple\\": \\"object\\"
  },
  \\"confirmed\\": \\"here it is\\"
};"
`;
